src/elements/ifc-client-link.ts

```diff
@@ -0,0 +1,58 @@
+import { Client, EnvData } from '../client';
+const PATH_ATTR = 'path';
+
+/**
+ * A custom element responsible for rendering an anchor element, turning a path
+ * within the current application to a full top-level url.
+ * It will be registered as `ifc-client-link` by a call to [[registerCustomElementsClient]] and
+ * should not be created directly. Instead add a `<ifc-client-link>` element to your
+ * markup or use `document.createElement('ifc-client-link')` after calling
+ * [[registerCustomElementsClient]].
+ */
+
+export default class IfcClientLinkElement extends HTMLElement {
+  private _client: Client;
+  private _environmentData: EnvData;
+  private _assignedRoute: string;
+
+  /** @internal */
+  constructor() {
+    super();
+    this._client = new Client({
+      hostOrigin: 'http://localhost:8080'
+    });
+    this._environmentData = {
+      locale: 'en-US',
+      hostRootUrl: 'http://localhost:3000/#/'
+    };
+    this._assignedRoute = '/app1';
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  static get observedAttributes() {
+    return PATH_ATTR;
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  public get environmentData() {
+    return this._environmentData;
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  public connectedCallback() {
+    const thisInnerHTML = this.innerHTML;
+    let path = this.getAttribute(PATH_ATTR) || 'about:blank';
+    const urlFromClientPath = this._client.urlFromClientPath.bind(this);
```

Comment:
You shouldn't need the `bind(this)` for this call, at least not once you're able to pass in an initialized client instance. You can just directly call `this._client.urlFromClientPath(path)`